easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -118,7 +118,7 @@ def configure_step(self, srcdir=None, builddir=None):
 
         setup_cmake_env(self.toolchain)
 
-        if builddir is None and self.cfg.get('separate_build_dir', False):
+        if builddir is None and self.cfg['separate_build_dir']:
```

Comment:
Please keep this as it was, just change the `False` to `True` (or, if you want to ensure this stays consistent, add a `DEFAULT_SEPARATE_BUILD_DIR` class constant and use that in both places.

We're careful with this because they may be easyblocks that derive from `CMakeMake` but blindly overwrite custom easyconfig parameters in `extra_options`, causing `'separate_build_dir'` to be undefined in `self.cfg`.

That may not be the case of any of the easyblocks in the central repository, but there may be other easyblocks out there that are less carefully constructed.

```python
f builddir is None and self.cfg.get('separate_build_dir', True):

Reply:
> but blindly overwrite custom easyconfig parameters in extra_options, causing separate_build_dir' to be undefined in self.cfg.

Wouldn't that be a bug (to do that)? IMO it is good that this would fail in that case.

There is also a usage of e.g. `self.cfg.get('configure_cmd')` which in that case (blindly overwriting) would return `None` and fail. Same for `build_type` and the property.

So I'd rather keep this change to make people fix their broken EasyBlocks (if there are any) unless you really want that

Reply:
fixed in 6695267